### DISCUSSION TITLE
Enable dynamic client-side forms

### DISCUSSION THREAD

**Discussion by pchemguy** [sqlpage/SQLPage#406] [Ideas]
Can you enable "formaction" on other editable fields?  

I want to implement essentially database table browser functionality. I want to show a form displaying data from a database table row. Now, I want to be able to show the "id" (type: number, step: 1) field, and update the form whenever id is changed. Similarly, I might also have some unique fields populated as drop-down option lists. When I change those, I want the form update automatically as well without the need to press a separate button. (I am aware of the table component, which is great, and I use it, but I am talking here about an alternative "row-detail" design.)

Perhaps, this requires Javascript + setting ONUPDATE event handlers, but I would prefer to have at least a template / sample project, so that I do not have to figure it all out by myself from scratch.  

Or maybe you can set it up automatically for a non-submit form field with "formaction" defined?

**Comment by lovasoa** [comment]
Hi ! I renamed the topic to avoid confusion with `formaction`, which is a different concept, not directly related to updating user interfaces dynamically on the client side.

This is something we could work on, but in the mean time, a good approach is to just have a "next" button that allows you to build your form progressively based on previous inputs. You can find an example for it on the official website: https://sql.ophir.dev/examples/multistep-form/ (the source code for it is [here](https://github.com/lovasoa/SQLpage/blob/main/examples/official-site/examples/multistep-form/index.sql))
